src/common/src/util/resource_util.rs

```diff
@@ -0,0 +1,625 @@
+
+    /// Returns the total number of cpu available as a float.
+    ///
+    /// If running in container, this function will return the cpu limit by the container. If not,
+    /// it will return the ```available_parallelism``` by the system. A panic will be invoked if
+    /// invoking process does not have permission to read appropriate values by
+    /// ```std::thread::available_parallelism``` or if the platform is not supported. The cgroup
+    /// mount point is assumed to be at /sys/fs/cgroup by default.
+    ///
+    /// [`with_capacity`]: #method.with_capacity
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    /// ``` ignore
+    /// let cpu_available = cpu::total_cpu_available();
+    /// ```
+    pub fn total_cpu_available() -> f32 {
+        if !super::runtime::is_linux_machine() || !super::runtime::is_running_in_container() {
+            return get_system_cpu();
+        }
+
+        get_container_cpu_limit(super::util::get_cgroup_version()).unwrap_or_else(get_system_cpu)
+    }
+
+    // Returns the CPU limit of the container.
+    fn get_container_cpu_limit(cgroup_version: Option<super::CgroupVersion>) -> Option<f32> {
+        if !super::util::is_controller_activated(super::Controller::Cpu) || cgroup_version.is_none()
+        {
+            return None;
+        }
+
+        match cgroup_version.unwrap() {
+            super::CgroupVersion::V1 => get_cpu_limit_v1(),
+            super::CgroupVersion::V2 => get_cpu_limit_v2(),
+        }
+    }
+
+    // Returns the total system cpu.
+    pub fn get_system_cpu() -> f32 {
+        match thread::available_parallelism() {
+            Ok(available_parallelism) => available_parallelism.get() as f32,
+            Err(e) => panic!("Platform is not supported, error: {}", e),
+        }
+    }
+
+    // Returns the CPU limit when cgroup_V1 is utilised.
+    fn get_cpu_limit_v1() -> Option<f32> {
+        let cpu_quota = match super::util::read_integer_from_file_path(&format!(
+            "{}{}",
+            super::DEFAULT_CGROUP_ROOT_HIERARCYHY,
+            V1_CPU_QUOTA_HIERARCHY
+        )) {
+            Some(quota_val) => quota_val,
+            None => return None,
+        };
+
+        let cpu_period = match super::util::read_integer_from_file_path(&format!(
+            "{}{}",
+            super::DEFAULT_CGROUP_ROOT_HIERARCYHY,
+            V1_CPU_PERIOD_HIERARCHY
+        )) {
+            Some(period_val) => period_val,
+            None => return None,
+        };
+        Some((cpu_quota as f32) / (cpu_period as f32))
+    }
+
+    // Returns the CPU limit when cgroup_V2 is utilised.
+    fn get_cpu_limit_v2() -> Option<f32> {
+        util::read_cgroup_v2_cpu_limit_from_file_path(&format!(
+            "{}{}",
+            super::DEFAULT_CGROUP_ROOT_HIERARCYHY,
+            V2_CPU_LIMIT_HIERARCHY
+        ))
+    }
+}
+
+mod util {
+    use std::fs;
+    use std::path::Path;
+    // Returns a cgroup version if it exists, else returns None.
```

Comment:
Add a blank line to separate the comments from the lines above.

```suggestion

    // Returns a cgroup version if it exists, else returns None.
```